src/overlays/actors/ovl_Boss_Hakugin/z_boss_hakugin.c

```diff
@@ -46,342 +113,3727 @@ ActorInit Boss_Hakugin_InitVars = {
+        this->rockEffects[i].timer = -1;
+    }
+
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->baseRotY = this->actor.shape.rot.y;
+    this->limbDrawFlags = 0xFFFFFFFF;
+
+    if (CHECK_EVENTINF(EVENTINF_INTRO_CS_WATCHED_GOHT)) {
+        SEQCMD_PLAY_SEQUENCE(SEQ_PLAYER_BGM_MAIN, 0, NA_BGM_BOSS | SEQ_FLAG_ASYNC);
+        this->actor.world.pos.x = 0.0f;
+        BossHakugin_SetupRun(this);
+    } else if (CHECK_EVENTINF(EVENTINF_ENTR_CS_WATCHED_GOHT)) {
+        this->iceAlpha = 255;
+        this->iceScaleY = 2.7f;
+        BossHakugin_SpawnLargeStalactiteWalls(this);
+        Animation_PlayOnce(&this->skelAnime, &gGohtThawAndBreakWallAnim);
+        this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+        BossHakugin_SetupFrozenBeforeFight(this);
+    } else {
+        this->iceAlpha = 255;
+        this->iceScaleY = 2.7f;
+        BossHakugin_SpawnLargeStalactiteWalls(this);
+        BossHakugin_SetupCutsceneStart(this);
+    }
+}
+
+void BossHakugin_Destroy(Actor* thisx, PlayState* play) {
+    BossHakugin* this = THIS;
+    s32 pad;
+    s32 i;
+
+    LightContext_RemoveLight(play, &play->lightCtx, this->lightNode);
+    Collider_DestroyJntSph(play, &this->bodyCollider);
+    Collider_DestroyCylinder(play, &this->iceCollider);
+
+    for (i = 0; i < GOHT_LIGHTNING_SEGMENT_COUNT; i++) {
+        Collider_DestroyTris(play, &this->lightningSegments[i].collider);
+    }
+
+    Collider_DestroySphere(play, &this->electricBallCollider);
+    AudioSfx_StopByPos(&this->sfxPos);
+    AudioSfx_StopByPos(&this->electricBallSfxPos);
+}
+
+s32 BossHakugin_Vec3fNormalize(Vec3f* vec) {
+    f32 magnitude = Math3D_Vec3fMagnitude(vec);
+
+    if (magnitude < 0.0001f) {
+        return false;
+    }
+
+    Math_Vec3f_Scale(vec, 1.0f / magnitude);
+    return true;
+}
+
+/**
+ * The intention behind this function is to call it repeatedly to gradually rotate `vector` to match the `targetVector`.
+ * Specifically, this function rotates `vector` around the axis perpendicular to both `vector` and `targetVector` (in
+ * other words, it rotates around the cross product). The amount by which to rotate `vector` per call is determined by
+ * `angleStep`; if rotating `vector` by this amount would cause it to overshoot the `targetVector`, then the rotation
+ * will be clamped such that the `vector` will reach its target exactly.
+ *
+ * This function assumes that both `vector` and `targetVector` are normalized unit vectors; if non-unit vectors are
+ * supplied, then this function will do nothing at all if the dot product of the two non-unit vectors is positive. It is
+ * also possible for this function to undershoot the `targetVector` if the angle between the `vector` and the
+ * `targetVector` is less than one degree; in this case, this function will return without doing any rotation.
+ */
+void BossHakugin_StepVector(Vec3f* vector, Vec3f* targetVector, f32 angleStep) {
```

Comment:
Since `vector` and `targetVector` should be norm vectors, what about naming them `norm` and `targetNorm`? Or maybe `vecNorm` and `targetVecNorm`?